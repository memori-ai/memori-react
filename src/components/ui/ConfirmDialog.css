/* Confirmation dialog styles */
.memori-confirm-dialog--backdrop {
    position: fixed;
    z-index: 50;
    background-color: rgba(0, 0, 0, 0.5);
    inset: 0;
  }
  
  .memori-confirm-dialog--container {
    position: fixed;
    z-index: 50;
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 1rem;
    inset: 0;
  }
  
  .memori-confirm-dialog--panel {
    width: 100%;
    max-width: 28rem;
    padding: 1.5rem;
    border-radius: 0.5rem;
    background-color: white;
    box-shadow: 0 20px 25px -5px rgba(0, 0, 0, 0.1), 0 10px 10px -5px rgba(0, 0, 0, 0.04);
  }
  
  .memori-confirm-dialog--title {
    margin-bottom: 0.75rem;
    font-size: 1.25rem;
    font-weight: 600;
  }
  
  .memori-confirm-dialog--message {
    margin-bottom: 1.5rem;
  }
  
  .memori-confirm-dialog--actions {
    display: flex;
    justify-content: flex-end;
    gap: 0.5rem;
  }