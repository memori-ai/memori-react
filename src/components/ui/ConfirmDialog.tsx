import { Dialog } from "@headlessui/react";

import { Transition } from "@headlessui/react";
import { FC } from "react";
import Button from "./Button";
import React from "react";
import "./ConfirmDialog.css";
// Custom confirm dialog component
const ConfirmDialog: FC<{
    isOpen: boolean;
    onClose: () => void;
    onConfirm: () => void;
    title: string;
    message: string;
    confirmText: string;
    cancelText: string;
  }> = ({ isOpen, onClose, onConfirm, title, message, confirmText, cancelText }) => {
    return (
      <Transition appear show={isOpen} as={React.Fragment}>
        <Dialog as="div" className="memori-confirm-dialog" onClose={onClose}>
          <div className="memori-confirm-dialog--backdrop" />
          <div className="memori-confirm-dialog--container">
            <Transition.Child
              as={React.Fragment}
              enter="ease-out duration-300"
              enterFrom="opacity-0 scale-95"
              enterTo="opacity-100 scale-100"
              leave="ease-in duration-200"
              leaveFrom="opacity-100 scale-100"
              leaveTo="opacity-0 scale-95"
            >
              <Dialog.Panel className="memori-confirm-dialog--panel">
                <Dialog.Title className="memori-confirm-dialog--title">{title}</Dialog.Title>
                <Dialog.Description className="memori-confirm-dialog--message">
                  {message}
                </Dialog.Description>
                <div className="memori-confirm-dialog--actions">
                  <Button  onClick={onClose}>
                    {cancelText}
                  </Button>
                  <Button primary onClick={onConfirm}>
                    {confirmText}
                  </Button>
                </div>
              </Dialog.Panel>
            </Transition.Child>
          </div>
        </Dialog>
      </Transition>
    );
  };


export default ConfirmDialog;