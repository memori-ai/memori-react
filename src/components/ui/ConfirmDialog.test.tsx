import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import ConfirmDialog from './ConfirmDialog';

beforeEach(() => {
  // @ts-ignore
  window.IntersectionObserver = jest.fn(() => ({
    observe: jest.fn(),
    unobserve: jest.fn(),
    disconnect: jest.fn(),
    takeRecords: jest.fn(),
  }));
});

it('renders ConfirmDialog unchanged', () => {
  const { container } = render(
    <ConfirmDialog
      isOpen={false}
      onClose={jest.fn()}
      onConfirm={jest.fn()}
      title="Test Title"
      message="Test Message"
      confirmText="Confirm"
      cancelText="Cancel"
    />
  );
  expect(container).toMatchSnapshot();
});

it('renders ConfirmDialog open unchanged', () => {
  const { container } = render(
    <ConfirmDialog
      isOpen={true}
      onClose={jest.fn()}
      onConfirm={jest.fn()}
      title="Test Title"
      message="Test Message"
      confirmText="Confirm"
      cancelText="Cancel"
    />
  );
  expect(container).toMatchSnapshot();
});

it('renders ConfirmDialog with custom title unchanged', () => {
  const { container } = render(
    <ConfirmDialog
      isOpen={true}
      onClose={jest.fn()}
      onConfirm={jest.fn()}
      title="Custom Title"
      message="Test Message"
      confirmText="Confirm"
      cancelText="Cancel"
    />
  );
  expect(container).toMatchSnapshot();
});

it('renders ConfirmDialog with custom message unchanged', () => {
  const { container } = render(
    <ConfirmDialog
      isOpen={true}
      onClose={jest.fn()}
      onConfirm={jest.fn()}
      title="Test Title"
      message="Custom confirmation message for testing purposes"
      confirmText="Confirm"
      cancelText="Cancel"
    />
  );
  expect(container).toMatchSnapshot();
});

it('renders ConfirmDialog with custom button text unchanged', () => {
  const { container } = render(
    <ConfirmDialog
      isOpen={true}
      onClose={jest.fn()}
      onConfirm={jest.fn()}
      title="Test Title"
      message="Test Message"
      confirmText="Yes, proceed"
      cancelText="No, go back"
    />
  );
  expect(container).toMatchSnapshot();
});

it('calls onClose when cancel button is clicked', () => {
  const onCloseMock = jest.fn();
  render(
    <ConfirmDialog
      isOpen={true}
      onClose={onCloseMock}
      onConfirm={jest.fn()}
      title="Test Title"
      message="Test Message"
      confirmText="Confirm"
      cancelText="Cancel"
    />
  );
  
  fireEvent.click(screen.getByText('Cancel'));
  expect(onCloseMock).toHaveBeenCalledTimes(1);
});

it('calls onConfirm when confirm button is clicked', () => {
  const onConfirmMock = jest.fn();
  render(
    <ConfirmDialog
      isOpen={true}
      onClose={jest.fn()}
      onConfirm={onConfirmMock}
      title="Test Title"
      message="Test Message"
      confirmText="Confirm"
      cancelText="Cancel"
    />
  );
  
  fireEvent.click(screen.getByText('Confirm'));
  expect(onConfirmMock).toHaveBeenCalledTimes(1);
});