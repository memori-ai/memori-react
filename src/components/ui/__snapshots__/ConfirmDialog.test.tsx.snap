// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ConfirmDialog open unchanged 1`] = `
<div>
  <div
    style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
  />
</div>
`;

exports[`renders ConfirmDialog unchanged 1`] = `<div />`;

exports[`renders ConfirmDialog with custom button text unchanged 1`] = `
<div>
  <div
    style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
  />
</div>
`;

exports[`renders ConfirmDialog with custom message unchanged 1`] = `
<div>
  <div
    style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
  />
</div>
`;

exports[`renders ConfirmDialog with custom title unchanged 1`] = `
<div>
  <div
    style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
  />
</div>
`;
